src/mozilla_sec_eia/models/sec10k/sec_output_table.py

```diff
@@ -0,0 +1,367 @@
+        merged_df["location_of_inc_ex21"].fillna("").str.lower().str.split()
+    )
+    # get the number of words overlapping between location of incorporation tokens
+    merged_df["loc_overlap"] = merged_df.apply(
+        lambda row: len(set(row["loc_tokens_sec"]) & set(row["loc_tokens_ex21"])),
+        axis=1,
+    )
+    # get the difference in report years
+    merged_df["report_year_diff"] = merged_df.apply(
+        lambda row: abs(int(row["report_year_sec"]) - int(row["report_year_ex21"])),
+        axis=1,
+    )
+    merged_df = merged_df.sort_values(
+        by=[
+            "company_name",
+            "location_of_inc_ex21",
+            "loc_overlap",
+            "report_year_diff",
+        ],
+        ascending=[True, True, False, True],
+    )
+    # Select the row with the highest loc overlap and nearest report years
+    # for each company name, location, and parent company record
+    closest_match_df = merged_df.groupby(
+        ["company_name", "location_of_inc_ex21", "parent_company_cik"], as_index=False
+    ).first()
+    ex21_with_cik_df = ex21_df.merge(
+        closest_match_df[
+            [
+                "company_name",
+                "parent_company_cik",
+                "location_of_inc_ex21",
+                "central_index_key",
+            ]
+        ].rename(columns={"location_of_inc_ex21": "location_of_inc"}),
+        how="left",
+        on=["company_name", "location_of_inc", "parent_company_cik"],
+    ).rename(columns={"central_index_key": "subsidiary_cik"})
+    # if a subsidiary doesn't have a CIK and has a null location
+    # but its company name was assigned a CIK (with a different location)
+    # then assign that CIK to the subsidiary
+    ex21_with_cik_df = ex21_with_cik_df.merge(
+        closest_match_df[["company_name", "central_index_key"]],
+        how="left",
+        on="company_name",
+    ).rename(columns={"central_index_key": "company_name_merge_cik"})
+    ex21_with_cik_df["subsidiary_cik"] = ex21_with_cik_df["subsidiary_cik"].where(
+        ~(ex21_with_cik_df.subsidiary_cik.isnull())
+        | ~(ex21_with_cik_df.location_of_inc.isnull()),
+        ex21_with_cik_df["company_name_merge_cik"],
+    )
+    ex21_with_cik_df = ex21_with_cik_df.drop(columns="company_name_merge_cik")
+    ex21_with_cik_df = ex21_with_cik_df.rename(
+        columns={"subsidiary_cik": "central_index_key"}
+    )
+    return ex21_with_cik_df
+
+
+def create_sec_company_id_for_ex21_subs(ex21_df: pd.DataFrame) -> pd.DataFrame:
+    """Create an sec_company_id for Ex. 21 subsidiaries.
+
+    This is a unique identifier string for Ex. 21 subsidiaries.
+    This ID is necessary for tracking subsidiaries who aren't ultimately
+    matched to a 10K filer company.
+    """
+    ex21_df.loc[:, "sec_company_id"] = (
+        ex21_df["parent_company_cik"]
+        + "_"
+        + ex21_df["company_name"]
+        + "_"
+        + ex21_df["location_of_inc"]
+    )
```

Comment:
I decided to just concatenate the strings instead of hashing them because then the ID stays more readable. Parent company + company name + location of inc should be unique because we drop duplicates on that subset.